base.js

```diff
@@ -143,10 +412,105 @@ module.exports = {
       },
     ],
 
+    // Disallow use of assignment in return statement
+    'no-return-assign': ['error', 'always'],
+
+    // Disallow redundant `return await`
+    'no-return-await': 'error',
+
+    // Disallow comparisons where both sides are exactly the same
+    'no-self-compare': 'error',
+
+    // -isallow use of comma operator
+    'no-sequences': 'error',
+
+    // Disallow use of `javascript:` urls.
+    'no-script-url': 'error',
+
+    // Disallow declaration of variables already declared in the outer scope
+    'no-shadow': 'error',
+
+    // Disallow template literal placeholder syntax in regular strings
+    'no-template-curly-in-string': 'error',
+
+    // Restrict what can be thrown as an exception
+    'no-throw-literal': 'error',
+
+    // Disallow useless string concatenation
+    'no-useless-concat': 'error',
+
+    // Disallow redundant return; keywords
+    'no-useless-return': 'error',
+
+    // Disallow use of undefined when initializing variables
+    'no-undef-init': 'error',
+
     'no-underscore-dangle': 'off',
 
+    // Disallow loops with a body that allows only one iteration
+    'no-unreachable-loop': [
+      'error',
+      {
+        ignore: [], // WhileStatement, DoWhileStatement, ForStatement, ForInStatement, ForOfStatement
+      },
+    ],
+
+    // Disallow use of variables before they are defined
+    'no-use-before-define': [
+      'error',
+      { functions: true, classes: true, variables: true },
+    ],
```

Comment:
thought(no action needed): these are the defaults, but it feels reasonable to spell them out here explicitly to be able to see right away what is explicitly allowed.